pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -72,21 +79,30 @@ type ClusterSpec struct {
 	Storage rookv1.StorageScopeSpec `json:"storage,omitempty"`
 
 	// The annotations-related configuration to add/set on each Pod related object.
+	// +kubebuilder:pruning:PreserveUnknownFields
 	Annotations rookv1.AnnotationsSpec `json:"annotations,omitempty"`
 
 	// The labels-related configuration to add/set on each Pod related object.
+	// +kubebuilder:pruning:PreserveUnknownFields
+	// +nullable
 	Labels rookv1.LabelsSpec `json:"labels,omitempty"`
 
 	// The placement-related configuration to pass to kubernetes (affinity, node selector, tolerations).
+	// +kubebuilder:pruning:PreserveUnknownFields
+	// +nullable
 	Placement rookv1.PlacementSpec `json:"placement,omitempty"`
```

Comment:
Does `nullable` have to go here, or can `nullable` be specified at `rookv1.PlacementSpec`? That would be a little nicer in that the APIs that use `rookv1.PlacementSpec` could avoid having to remember to put `nullable`.

Also what about for `kubebuilder:pruning:PreserveUnknownFields`?